### PR TITLE
default the preview_max_memory to a half of the max php memory

### LINKED ISSUE #31284: [Bug]: Nextcloud cannot create previews for jpg images 34 MP or larger
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [x] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Nextcloud cannot create previews for jpg images 34 MP or larger.

### Steps to reproduce

1. Start a fresh docker container with any of these nextcloud versions: 23.0.2 or  or  22.2.5   or  21.0.9 
2. Go to files and drop the following images (recommended: one by one)
   * https://files.bulitta.com/dynsrc/2022-02-20-nextcloud-bug-large-images/30_MP_-_JPG-Quality_80.jpg
   * https://files.bulitta.com/dynsrc/2022-02-20-nextcloud-bug-large-images/32_MP_-_JPG-Quality_80.jpg
   * https://files.bulitta.com/dynsrc/2022-02-20-nextcloud-bug-large-images/34_MP_-_JPG-Quality_10.jpg
   * https://files.bulitta.com/dynsrc/2022-02-20-nextcloud-bug-large-images/34_MP_-_JPG-Quality_80.jpg
   * https://files.bulitta.com/dynsrc/2022-02-20-nextcloud-bug-large-images/36_MP_-_JPG-Quality_80.jpg
   * https://files.bulitta.com/dynsrc/2022-02-20-nextcloud-bug-large-images/38_MP_-_JPG-Quality_80.jpg
   * https://files.bulitta.com/dynsrc/2022-02-20-nextcloud-bug-large-images/40_MP_-_JPG-Quality_80.jpg
   * https://files.bulitta.com/dynsrc/2022-02-20-nextcloud-bug-large-images/45_MP_-_JPG-Quality_1.jpg
   * https://files.bulitta.com/dynsrc/2022-02-20-nextcloud-bug-large-images/45_MP_-_JPG-Quality_50.jpg
   * https://files.bulitta.com/dynsrc/2022-02-20-nextcloud-bug-large-images/45_MP_-_JPG-Quality_80.jpg
   * https://files.bulitta.com/dynsrc/2022-02-20-nextcloud-bug-large-images/45_MP_-_JPG-Quality_90.jpg
3. Wait after each dropped picture for the CPU to become idle
4. Refresh the browser
5. Now you should get previews (thumbnails + larger previews) for all the dropped pictures, but actually you only get previews for pictures smaller than 34 MP, no matter how many bytes the jpg file is: 
2022-02-20_19_03_46-Photos_-_Dateien_-_Nextcloud
2022-02-20_19_04_44-34_MP_-_JPG-Quality_10 jpg_-_Nextcloud

### Expected behavior

Nextcloud can create previews for ALL jpg images, even 34 MP or larger.

### Installation method

Official Docker image

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.0

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### Are you using the Nextcloud Server Encryption module?

_No response_

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
root@odroid-DMZ-webserver:/mnt/rootfs-sd/docker/webserver/mountpoints/var-www-html/sd-card# docker exec -it -u www-data nextcloud-test php occ config:list system
{
    "system": {
        "htaccess.RewriteBase": "\/",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "apps_paths": [
            {
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/custom_apps",
                "url": "\/custom_apps",
                "writable": true
            }
        ],
        "overwritehost": "pubcloud-test.bulitta.com",
        "overwriteprotocol": "http",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "pubcloud-test.bulitta.com",
            "nextcloud"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "23.0.2.1",
        "overwrite.cli.url": "http:\/\/localhost",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***"
    }
}
```

### List of activated Apps

```shell
root@odroid-DMZ-webserver:/mnt/rootfs-sd/docker/webserver/mountpoints/var-www-html/sd-card# docker exec -it -u www-data nextcloud-test php occ app:list
Enabled:
  - accessibility: 1.9.0
  - activity: 2.15.0
  - circles: 23.0.1
  - cloud_federation_api: 1.6.0
  - comments: 1.13.0
  - contactsinteraction: 1.4.0
  - dashboard: 7.3.0
  - dav: 1.21.0
  - federatedfilesharing: 1.13.0
  - federation: 1.13.0
  - files: 1.18.0
  - files_pdfviewer: 2.4.0
  - files_rightclick: 1.2.0
  - files_sharing: 1.15.0
  - files_trashbin: 1.13.0
  - files_versions: 1.16.0
  - files_videoplayer: 1.12.0
  - firstrunwizard: 2.12.0
  - logreader: 2.8.0
  - lookup_server_connector: 1.11.0
  - nextcloud_announcements: 1.12.0
  - notifications: 2.11.1
  - oauth2: 1.11.0
  - password_policy: 1.13.0
  - photos: 1.5.0
  - privacy: 1.7.0
  - provisioning_api: 1.13.0
  - recommendations: 1.2.0
  - serverinfo: 1.13.0
  - settings: 1.5.0
  - sharebymail: 1.13.0
  - support: 1.6.0
  - survey_client: 1.11.0
  - systemtags: 1.13.0
  - text: 3.4.0
  - theming: 1.14.0
  - twofactor_backupcodes: 1.12.0
  - updatenotification: 1.13.0
  - user_status: 1.3.1
  - viewer: 1.7.0
  - weather_status: 1.3.0
  - workflowengine: 2.5.0
Disabled:
  - admin_audit
  - encryption
  - files_external
  - user_ldap
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
{"reqId":"FQqMvdjwHvM3de3F3r2o","level":3,"time":"2022-02-20T18:01:23+00:00","remoteAddr":"172.29.0.3","user":"admin","app":"core","method":"GET","url":"/core/preview?fileId=221&c=01e189d81ed2a8337fc740390119010e&x=313&y=313&forceIcon=0&a=0","message":"OC_Image::fixOrientation(): No image loaded","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0","version":"23.0.2.1"}
{"reqId":"jv5ECJZaMVGWFjNrvk6s","level":3,"time":"2022-02-20T18:01:36+00:00","remoteAddr":"172.29.0.3","user":"admin","app":"core","method":"GET","url":"/core/preview?fileId=232&c=13f7ef88c690455ed569ed80b2b22149&x=313&y=313&forceIcon=0&a=0","message":"OC_Image::fixOrientation(): No image loaded","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0","version":"23.0.2.1"}
{"reqId":"NMtE1ulnGdqPE01frb6p","level":3,"time":"2022-02-20T18:01:46+00:00","remoteAddr":"172.29.0.3","user":"admin","app":"core","method":"GET","url":"/core/preview?fileId=243&c=585af979e49838250c46560643d76917&x=313&y=313&forceIcon=0&a=0","message":"OC_Image::fixOrientation(): No image loaded","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0","version":"23.0.2.1"}
{"reqId":"u79kH1hK81nxLIRgFMYJ","level":3,"time":"2022-02-20T18:01:54+00:00","remoteAddr":"172.29.0.3","user":"admin","app":"core","method":"GET","url":"/core/preview?fileId=254&c=9e65626e725cfd1ca934904f1cdbce01&x=313&y=313&forceIcon=0&a=0","message":"OC_Image::fixOrientation(): No image loaded","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0","version":"23.0.2.1"}
{"reqId":"rw0n8jaKYU599BWvY4WO","level":3,"time":"2022-02-20T18:01:58+00:00","remoteAddr":"172.29.0.3","user":"admin","app":"core","method":"GET","url":"/core/preview?fileId=262&c=47e65e01e2c7296efe05301a5c9e0d23&x=313&y=313&forceIcon=0&a=0","message":"OC_Image::fixOrientation(): No image loaded","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0","version":"23.0.2.1"}
{"reqId":"xxaPzD3STgBzmgIwUL0i","level":3,"time":"2022-02-20T18:02:04+00:00","remoteAddr":"172.29.0.3","user":"admin","app":"core","method":"GET","url":"/core/preview?fileId=270&c=e057c1c9763a77f3fa0f9fa0a87438b7&x=313&y=313&forceIcon=0&a=0","message":"OC_Image::fixOrientation(): No image loaded","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0","version":"23.0.2.1"}
{"reqId":"KEYJyBNjDv4FKnUtcacA","level":3,"time":"2022-02-20T18:02:11+00:00","remoteAddr":"172.29.0.3","user":"admin","app":"core","method":"GET","url":"/core/preview?fileId=281&c=e0becdf3db23a5298aad35f08f73377a&x=313&y=313&forceIcon=0&a=0","message":"OC_Image::fixOrientation(): No image loaded","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0","version":"23.0.2.1"}
{"reqId":"LOs11UREghcw3UFaAq2h","level":3,"time":"2022-02-20T18:02:18+00:00","remoteAddr":"172.29.0.3","user":"admin","app":"core","method":"GET","url":"/core/preview?fileId=294&c=d596b85c4b0bb2ea6b462f80485afee5&x=313&y=313&forceIcon=0&a=0","message":"OC_Image::fixOrientation(): No image loaded","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0","version":"23.0.2.1"}
{"reqId":"v3j1UTlcCiFfdJgcKq0K","level":3,"time":"2022-02-20T18:03:01+00:00","remoteAddr":"172.29.0.3","user":"admin","app":"core","method":"GET","url":"/core/preview?fileId=302&c=887a09279d92916fa2bcb42ba002ef8a&x=313&y=313&forceIcon=0&a=0","message":"OC_Image::fixOrientation(): No image loaded","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0","version":"23.0.2.1"}
{"reqId":"egpvqvm1klWc35wFaPTq","level":3,"time":"2022-02-20T18:04:38+00:00","remoteAddr":"172.29.0.3","user":"admin","app":"core","method":"GET","url":"/core/preview?fileId=221&x=4096&y=2160&a=true","message":"OC_Image::fixOrientation(): No image loaded","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0","version":"23.0.2.1"}
{"reqId":"Wk320Ya5KyJECV6EHLEv","level":3,"time":"2022-02-20T18:04:38+00:00","remoteAddr":"172.29.0.3","user":"admin","app":"core","method":"GET","url":"/core/preview?fileId=302&x=4096&y=2160&a=true","message":"OC_Image::fixOrientation(): No image loaded","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0","version":"23.0.2.1"}
{"reqId":"slsCw5TkhfsCFJjz3VaY","level":3,"time":"2022-02-20T18:04:38+00:00","remoteAddr":"172.29.0.3","user":"admin","app":"core","method":"GET","url":"/core/preview?fileId=232&x=4096&y=2160&a=true","message":"OC_Image::fixOrientation(): No image loaded","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0","version":"23.0.2.1"}
{"reqId":"8vtlf485QD61cDLFlvf2","level":3,"time":"2022-02-20T18:04:41+00:00","remoteAddr":"172.29.0.3","user":"admin","app":"core","method":"GET","url":"/core/preview?fileId=302&x=4096&y=2160&a=true","message":"OC_Image::fixOrientation(): No image loaded","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0","version":"23.0.2.1"}
{"reqId":"45Q4RwXQtgLdn684C35H","level":3,"time":"2022-02-20T18:04:41+00:00","remoteAddr":"172.29.0.3","user":"admin","app":"core","method":"GET","url":"/core/preview?fileId=221&x=4096&y=2160&a=true","message":"OC_Image::fixOrientation(): No image loaded","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0","version":"23.0.2.1"}
```

### Additional info

I already tried to figure out the differences between the last working version 20.0.14 , the first not working version 21.0.9 and the currently latest version 23.0.2 using phpinfo.

The candidate, which looks must interesting for me is the version change of the **imagick module** :

|                            | nextcloud:20 | nextcloud:21 | nextcloud:23 |
| -------------------------- | ------------ | ------------ | ------------ |
| Previews                   | OK           | FAIL         | FAIL         |
| PHP Version                | 7.4.26       | 7.4.28       | 8.0.16       |
| **imagick module version** | **3.5.1**    | **3.7.0**    | **3.7.0**    |
| memory_limit               | 512M         | 512M         | 512M         |

As I am not a php professional, I tried out only a few simple things like this:

```php
<?php
/* Read the image */
#$im = new Imagick("/var/www/html/data/admin/files/Photos/2020-07-20_12-26-29-_U2A6017-A.jpg"); // small
$im = new Imagick("/var/www/html/data/admin/files/Photos/2021-08-08_18-14-44-_N6A2872-A.jpg");  // large

/* Thumbnail the image */
$im->thumbnailImage(200, null);

/* Output the image*/
$im->writeImages( 'thumbnail.jpg', true );
?>
```

Unfortunately imagick worked for ALL pictures in this test snippet code.

Any ideas how to fix this?

### PR DESCRIPTION
**Pull request by szaimen**
Close https://github.com/nextcloud/server/issues/31284

### CODE REVIEW

File: tests/lib/ImageTest.php [hunk 1]
```diff
@@ -144,8 +144,8 @@ public function testData() {
 			->willReturn(null);
 		$config->expects($this->once())
 			->method('getSystemValueInt')
-			->with('preview_max_memory', 128)
-			->willReturn(128);
+			->with('preview_max_memory', 384)
```
Comment (kesselb) [inline tests/lib/ImageTest.php:147]: The value depends on your machine. I fear we are going to introduce the next "wonky" test ;) Could you try to set the memory limit for this test to a fixed value? 

ini_set('memory_limit', '256M');

Best would be to mock the MemoryInfo class but that's a bit tricky here.

Comment (szaimen) [reply tests/lib/ImageTest.php:147]: > Could you try to set the memory limit for this test to a fixed value?
> 
> ini_set('memory_limit', '256M');

Good idea, thanks! :)

**Comment by kesselb** [comment]
https://github.com/nextcloud/server/blob/5848a3673a47e3b0bd8f4d5103b535215bdcf5c9/lib/private/legacy/OC_Image.php#L597

A higher log level could also help. 

Auto-calculating preview_max_memory is a bit too much magic for me. 
I prefer to improve our documentation instead of yet another runtime check for a static value.

**Comment by szaimen** [comment]
> Auto-calculating preview_max_memory is a bit too much magic for me.

All right. Then lets increase the default to 256 and also increase the log entry to info which should fix the issue as well imho.

**Comment by kesselb** [comment]
> All right. Then lets increase the default to 256 and also increase the log entry to info which should fix the issue as well

Your choice ;) I just shared my concerns.

The issue with info is that most people use warning as default log level.

**Comment by szaimen** [comment]
Superseded by https://github.com/nextcloud/server/pull/35856

**Pull request closed without merging**